src/routes/user.ts

```diff
@@ -0,0 +1,88 @@
+import { FastifyInstance, FastifyReply, FastifyRequest } from 'fastify';
+import { Type, Static } from '@sinclair/typebox';
+import { knexController } from '../database/database';
+
+const userAddedResponse = Type.Object({
+  status: Type.String(),
+});
+
+export type UserAddedResponse = Static<typeof userAddedResponse>;
+
+const schema = {
+  description: 'Add new user or return error if add is not successful',
+  summary: 'Add user',
+  tags: ['User'],
+  body: {
+    type: 'object',
+    required: ['email', 'forename', 'surname', 'password'],
+    properties: {
+      email: {
+        type: 'string',
+      },
+      forename: {
+        type: 'string',
+      },
+      surname: {
+        type: 'string',
+      },
+      password: {
+        type: 'string',
+      },
+    },
+  },
+  response: {
+    201: {
+      description: 'User added',
+      type: 'null',
+    },
+    409: {
+      description: 'Email already in use',
+      type: 'null',
+    },
+    500: {
+      description: 'Server error',
+      type: 'null',
+    },
+  },
+};
+
+const handler = async (
+  req: FastifyRequest,
+  reply: FastifyReply
+): Promise<void> => {
+  if (req.body) {
```

Comment:
Fastify automatically checks that the body exists and the required parameters are part of the body. No need to do double checking

Reply:
Typescript compiler oli demoillessa eri mieltä :D

Reply:
Auttanee kun kattellaan noi tyypit kuntoon

Reply:
Sun pitää toki kertoa sille requestille minkä muotonen se on. Menisköhän ihan vaan FastifyRequest<UserCreationBody> tjsp